### PR TITLE
[#5] Create OrchestratorStorage with service loading from runtime

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

**Comment by jbonofre** [comment]
@niuzhenguo @iskey FYI, I updated the GitHub action workflow to build/check on each pull request (see the check tab).

File: services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/MemoryOrchestratorStorageImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.eclipse.osc.orchestrator;
+
+import org.apache.karaf.minho.boot.spi.Service;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+public class MemoryOrchestratorStorageImpl implements OrchestratorStorage, Service {
```
Comment (niuzhenguo) [inline services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/MemoryOrchestratorStorageImpl.java:8]: Should we keep this memory storage way? As discussed last time, we are supposed to have a local file storage impl at least.

Comment (jbonofre) [reply services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/MemoryOrchestratorStorageImpl.java:8]: Yes. Good point. Let me update the PR with local file system storage and provide another distributed one (powered by Hazelcast for instance) as example.

**Comment by jbonofre** [comment]
I'm resuming my work on this one, implementing a local filesystem storage by default (instead of memory).

**Comment by jbonofre** [comment]
By the way, just a note: the storage implementation could be part of specific orchestrator plugin. For instance, within HuaweiCloud plugin, we can provide a orchestrator storage implementation based on etcd, etc.

**Comment by niuzhenguo** [comment]
> By the way, just a note: the storage implementation could be part of specific orchestrator plugin. For instance, within HuaweiCloud plugin, we can provide a orchestrator storage implementation based on etcd, etc.

Yes, maybe an OBS storage implementation on Huawei Cloud and Swift on OpenStack.

**Comment by jbonofre** [comment]
@niuzhenguo I updated this PR with:
1. Use properties with local file storage by default instead of in-memry
2. Add Swift based orchestrator storage skeleton in the openstack plugin

**Comment by jbonofre** [comment]
@niuzhenguo @iskey I added openstack authentication in this PR by mistake. If it's OK for you, I will merge this PR and create a new one with moving forward on openstack plugin. Thoughts ?

File: services/openstack/src/main/java/org/eclipse/osc/orchestrator/plugin/openstack/OpenstackOrchestratorPlugin.java [hunk 2]
```diff
@@ -1,22 +1,90 @@
...
+     * @return the token
+     * @throws Exception if the authentcation fails
+     */
+    private String authenticate() throws Exception {
+        log.info("Authenticate on " + authenticationUrl);
+        // TODO support proxy
+        HttpURLConnection connection = (HttpURLConnection) new URL(authenticationUrl).openConnection();
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("Accept", "application/jsom");
```
Comment (niuzhenguo) [inline services/openstack/src/main/java/org/eclipse/osc/orchestrator/plugin/openstack/OpenstackOrchestratorPlugin.java:70]:
```suggestion
        connection.setRequestProperty("Accept", "application/json");
```

**Review by niuzhenguo — COMMENTED** [review]
LGTM, suggest to use OpenStack Java sdk to do authentication in future, but it's OK for now as a POC implementation.

**Comment by iskey** [comment]
LGTM, thanks!

File: services/openstack/src/main/java/org/eclipse/osc/orchestrator/plugin/openstack/OpenstackOrchestratorPlugin.java (hunk 2 above)
Comment (jbonofre) [reply services/openstack/src/main/java/org/eclipse/osc/orchestrator/plugin/openstack/OpenstackOrchestratorPlugin.java:70]: Oh good catch, let me fix that.

**Comment by jbonofre** [comment]
@niuzhenguo yes I plan to use openstack java sdk,  it's just skeleton for now.

**Review by iskey — APPROVED** [review]

**Pull request merged**